### DISCUSSION TITLE
Tps is slow down to 16 when some other program is running.

### DISCUSSION THREAD

**Discussion by jhooc77** [Minestom/Minestom#1349] [Help]
Anyone know about why minestom server slow down to tps 16 when some other program is running?
It slows down to exactly 16.
When I turn off the program, It took a few seconds to get back to normal.
It was a light program like Discord or Internet Explorer, and the cpu was high performance enough, so it's not a performance problem.
Average mspt was lower than 1, even max mspt wasn't over then 10.
Only tps was slow down.

This value is the result of the Spark, but I don't think it's a problem with the Spark because it was slow enough to feel it even when I didn't used the Spark.
I couldn't check all thread of the minestom, but the things I made through the scheduler slowed down(async execution too).
This may be a matter for Scheduler, but I don't know the details.
I don't know if it's Java's problem or what. I won't run another program when I actually use it, but I want to know the cause because there might be other problems.
![image](https://user-images.githubusercontent.com/41156145/185759229-b86399b6-6dd0-49af-acb7-eeb65a534d5c.png)

**Comment by HSGamer** [comment]
`spark` is not optimal to check performance in this case.
The only thing it can be used is to check profiler for any lag spikes that can be optimized. That's the original reason why I ported it to Minestom.
If you want to go that full performance check, use a proper profiler like JProfiler, VisualVM or async-profiler.
